File: rust/src/models/grouped_usage.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct GroupedUsage {
    pub dimensions: std::collections::HashMap<String, String>,

    pub value: rust_decimal::Decimal,
}

impl GroupedUsage {
    pub fn new(
        dimensions: std::collections::HashMap<String, String>,
        value: rust_decimal::Decimal,
    ) -> Self {
        Self { dimensions, value }
    }
}

File: typescript/src/models/groupedUsage.ts
// this file is @generated

export interface GroupedUsage {
  dimensions: { [key: string]: string };

  value: string;
}

export const GroupedUsageSerializer = {
  _fromJsonObject(object: any): GroupedUsage {
    return {
      dimensions: object["dimensions"],
      value: object["value"],
    };
  },

  _toJsonObject(self: GroupedUsage): any {
    return {
      dimensions: self.dimensions,
      value: self.value,
    };
  },
};
